tensorflow/python/ops/nn_impl.py

```diff
@@ -1046,13 +1054,19 @@ class biases.
 
     # Construct output logits and labels. The true labels/logits start at col 0.
     out_logits = array_ops.concat([true_logits, sampled_logits], 1)
-    # true_logits is a float tensor, ones_like(true_logits) is a float tensor
-    # of ones. We then divide by num_true to ensure the per-example labels sum
-    # to 1.0, i.e. form a proper probability distribution.
-    out_labels = array_ops.concat([
-        array_ops.ones_like(true_logits) / num_true,
-        array_ops.zeros_like(sampled_logits)
-    ], 1)
+    if labels_as_indices:
+      # We want each row of labels to be the indices of the targets, which
+      # start at col 0 and end at col num_true-1.
+      out_labels = gen_array_ops.tile(
```

Comment:
i guess this should only be used for num_true == 1, right?  can you add a check and corresponding documentation?

alternatively you could just move this code to sparse_sampled_softmax_loss and revert this.

Reply:
@drpngx just checking in because this has been sitting for some time. I responded to @ebrevdo 's review with the above comment. Thanks!